plugins/inputs/tail/tail.go

```diff
@@ -205,7 +228,20 @@ func (t *Tail) Stop() {
 	t.Lock()
 	defer t.Unlock()
 
+	offsetsMutex.Lock()
+	defer offsetsMutex.Unlock()
```

Comment:
This seems too coarse of locking, all the tail plugins will need to stop one at a time.  Let's wrap only the write to the offsets.

Reply:
Since we'll have a local copy of `offsets`, we could delay locking and copying to the global `offsets` until after `wg.Wait()`